### PR TITLE
test: migrate tests

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Task [link](https://textkernel.atlassian.net/browse/ONEUI-407)

### CODE REVIEW

File: src/components/Input/Input.tsx [hunk 1]
```diff
@@ -44,6 +45,7 @@ export const Input = React.forwardRef<HTMLInputElement, Props>(
                 disabled={disabled}
                 value={value}
                 data-lpignore={isLastPassDisabled}
+                role="textbox"
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/Input/Input.tsx:48]: Is the `input` present in the DOM? What error you get / what can be found? If you do add it here, do you find it?

[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/Input/Input.tsx:48]: If i remove role="textbox" for Input.tsx and then run test it shows 
image

**Comment by PavelKirvelTextkernel** [comment]
I'm not sure about role names

**Comment by PavelKirvelTextkernel** [comment]
I'm not sure about role names

File: src/components/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -61,10 +67,12 @@ export const Checkbox = React.forwardRef<HTMLInputElement, Props>(
...
                             height="10px"
                             viewBox={CHECKBOX_VIEWBOX}
+                            role="img"
+                            aria-label={indeterminate ? lineLabel : polylineLabel}
                         >
-                            {indeterminate && <line x1="2" y1="6" x2="8" y2="6" />}
-                            {!indeterminate && (checked === undefined || checked === true) && (
-                                <polyline points="1.5 6 3.5 9 8 3" />
+                            {indeterminate && <line x1="2" y1="6" x2="8" y2="6" role={lineLabel} />}
+                            {!indeterminate && (checked === undefined || checked) && (
```
Comment (PavelKirvelTextkernel) [inline src/components/Checkbox/Checkbox.tsx:74]: I'm not sure about this approach. Maybe better rename labels to roles ?

Comment (antipin) [reply src/components/Checkbox/Checkbox.tsx:77]: sounds good, less confusing. Especially having that you used `ariaRole` in `ColorValue.tsx`.

Let's rename

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx [hunk 3]
```diff
@@ -1,88 +1,128 @@
...
+        it('should render the checked svg when checked is true and indeterminate false', () => {
+            const newProps = {
+                id: 'c5',
+                indeterminate: false,
+            };
+
+            rerenderView(newProps);
+
+            expect(view.container).toMatchSnapshot();
+            expect(screen.getByRole('img', { name: polylineLabel })).toBeInTheDocument();
```
Comment (antipin) [inline src/components/Checkbox/__tests__/Checkbox.spec.tsx:98]: `.toBeVisisble()` may be better?

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx [hunk 4]
```diff
@@ -1,88 +1,128 @@
...
+        it('should render the indeterminate svg when checked and indeterminate are true', () => {
+            const newProps = {
+                id: 'c6',
+            };
+
+            rerenderView(newProps);
+
+            expect(view.container).toMatchSnapshot();
+            expect(screen.queryByRole('img', { name: polylineLabel })).not.toBeInTheDocument();
+            expect(screen.getByRole('img', { name: lineLabel })).toBeInTheDocument();
```
Comment (antipin) [inline src/components/Checkbox/__tests__/Checkbox.spec.tsx:111]: .toBeVisisble() may be better?

File: src/components/Text/MarkedText/__tests__/MarkedText.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        );
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getAllByRole('contentinfo')).toHaveLength(3);
+    });
+
+    it('should mark special characters', () => {
+        view = render(
+            // eslint-disable-next-line react/no-unescaped-entities
+            <MarkedText marker=".?!#$%^&()[]:;\?''">.?!#$%^&()[]:;\?''</MarkedText>
```
[the remark this answers is not in the document]
Comment (antipin) [reply src/components/Text/MarkedText/__tests__/MarkedText.spec.tsx:43]: That makes sense

File: src/components/AutosuggestDeprecated/ItemTag/__tests__/ItemTag.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.queryByRole('button')).not.toBeInTheDocument();
+    });
+
+    it('should add delete button if onClick is defined', () => {
+        const onClickMock = jest.fn();
+        view = render(<ItemTag onClick={onClickMock}>tag</ItemTag>);
+
+        expect(screen.getByRole('button')).toBeInTheDocument();
```
Comment (antipin) [inline src/components/AutosuggestDeprecated/ItemTag/__tests__/ItemTag.spec.tsx:21]: `.toBeVisible();` ?

File: src/components/Themeroller/ThemeTuner/ItemValue/ColorValue/__tests__/ColorValue.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+describe('ColorValue component', () => {
+    const item = { value: '#fffff' } as ThemeColorItem;
+    const inputValue = { value: '#00000' } as ThemeColorItem;
+    const role = 'colorselector';
+
+    let view: RenderResult;
+
+    it('should render component correctly', () => {
+        const onChangeMock = jest.fn();
+        view = render(<ColorValue item={item} onChange={onChangeMock} ariaRole={role} />);
```
Comment (antipin) [inline src/components/Themeroller/ThemeTuner/ItemValue/ColorValue/__tests__/ColorValue.spec.tsx:16]: `onChangeMock` is not really used, just pass `onChange={jest.fn()}`

File: src/components/Themeroller/ThemeTuner/ItemValue/NumberValue/__tests__/NumberValue.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+describe('NumberValue component', () => {
+    const item = { value: '100' } as ThemeNumberItem;
+    const inputValue = { value: '200' } as ThemeNumberItem;
+    const role = 'numberselector';
+
+    let view: RenderResult;
+
+    it('should render component correctly', () => {
+        const onChangeMock = jest.fn();
+        view = render(<NumberValue item={item} onChange={onChangeMock} ariaRole={role} />);
```
Comment (antipin) [inline src/components/Themeroller/ThemeTuner/ItemValue/NumberValue/__tests__/NumberValue.spec.tsx:16]: `onChange={onChangeMock} -> onChange={jest.fn()}`

**Review by antipin — CHANGES_REQUESTED** [review]

File: src/components/Themeroller/ThemeTuner/ItemValue/ColorValue/__tests__/ColorValue.spec.tsx (hunk 7 above)
Comment (PavelKirvelTextkernel) [reply src/components/Themeroller/ThemeTuner/ItemValue/ColorValue/__tests__/ColorValue.spec.tsx:16]: :+1:

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx (hunk 3 above)
Comment (PavelKirvelTextkernel) [reply src/components/Checkbox/__tests__/Checkbox.spec.tsx:98]: :+1:

File: src/components/Checkbox/__tests__/Checkbox.spec.tsx (hunk 4 above)
Comment (PavelKirvelTextkernel) [reply src/components/Checkbox/__tests__/Checkbox.spec.tsx:111]: :+1:

File: src/components/AutosuggestDeprecated/ItemTag/__tests__/ItemTag.spec.tsx (hunk 6 above)
Comment (PavelKirvelTextkernel) [reply src/components/AutosuggestDeprecated/ItemTag/__tests__/ItemTag.spec.tsx:21]: :+1:

File: src/components/Themeroller/ThemeTuner/ItemValue/NumberValue/__tests__/NumberValue.spec.tsx (hunk 8 above)
Comment (PavelKirvelTextkernel) [reply src/components/Themeroller/ThemeTuner/ItemValue/NumberValue/__tests__/NumberValue.spec.tsx:16]: :+1:

**Comment by PavelKirvelTextkernel** [comment]
@antipin I've pushed changes, could you check it ?

**Review by antipin — APPROVED** [review]
Looks good!

**Pull request merged**